README.md

```diff
@@ -28,19 +28,19 @@ The `-p` is short for [publish](https://docs.docker.com/network/#published-ports
 
 ## Stages
 
-The build is [multi-stage](https://docs.docker.com/build/building/multi-stage/) with several kinds of stages:
-
-- `START` The very first stage, which run `ag0` instead of `agd` as the other layers do. (This was the version of `agd` before JS VM.)
-- `PREPARE` For upgrade proposals: submits the proposal, votes on it, runs to halt for the next stage
-- `EXECUTE` For upgrade proposals: starts `agd` with the new SDK, letting its upgrade handler upgrade the chain
-- `EVAL` For core-eval proposals: submits the proposal, votes on it, and begin executing. Does not guarantee the eval will finish but does wait several blocks to give it a chance.
-
-All proposals then have two additional stages:
-
-- `USE` Perform actions to update the chain state, to persist through the chain history. E.g. adding a vault that will be tested again in the future.
-- `TEST` Test the chain state and perform new actions that should not be part of history. E.g. adding a contract that never was on Mainnet.
-
-The `TEST` stage does not RUN as part of the build. It only copies test files into the image and defines the ENTRYPOINT. CI runs those entrypoints to execute the tests.
+The build is [multi-stage](https://docs.docker.com/build/building/multi-stage/) with several kinds of stages.
+
+| Stage Type | Scope | Description |
+|--------|--------|--------|
+| START | base | The very first stage, which run `ag0` instead of `agd` as the other layers do. (This was the version of `agd` before JS VM.) |
+| PREPARE | chain software upgrade proposals | submits the proposal, votes on it, runs to halt for the next stage |
+| EXECUTE | chain software upgrade proposals | starts `agd` with the new SDK, letting its upgrade handler upgrade the chain |
+| EVAL | other proposals | submits the proposal, votes on it, and begin executing. Does not guarantee the eval will finish but does wait several blocks to give it a chance. |
+| USE | proposals | Perform actions to update the chain state, to persist through the chain history. E.g. adding a vault that will be tested again in the future. |
```

Comment:
consider "all proposals".

also consider referencing the flow diagram: https://github.com/Agoric/agoric-3-proposals/tree/main/packages/synthetic-chain#design

Reply:
Moved the flow diagram to the top-level README.